homeassistant/components/prosegur/alarm_control_panel.py

```diff
@@ -0,0 +1,79 @@
+"""Support for Prosegur alarm control panels."""
+import logging
+
+from pyprosegur.auth import Auth
+from pyprosegur.installation import Installation, Status
+
+import homeassistant.components.alarm_control_panel as alarm
+from homeassistant.components.alarm_control_panel import (
+    SUPPORT_ALARM_ARM_AWAY,
+    SUPPORT_ALARM_ARM_HOME,
+)
+from homeassistant.const import (
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_DISARMED,
+)
+
+from . import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STATE_MAPPING = {
+    Status.DISARMED: STATE_ALARM_DISARMED,
+    Status.ARMED: STATE_ALARM_ARMED_AWAY,
+    Status.PARTIALLY: STATE_ALARM_ARMED_HOME,
+    Status.ERROR_PARTIALLY: STATE_ALARM_ARMED_HOME,
+}
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the Prosegur alarm control panel platform."""
+    async_add_entities(
+        [ProsegurAlarm(entry.data["contract"], hass.data[DOMAIN][entry.entry_id])]
+    )
```

Comment:
This calls the update before adding to hass

```suggestion
    async_add_entities(
        [ProsegurAlarm(entry.data["contract"], hass.data[DOMAIN][entry.entry_id])], 
        update_before_add=True,
    )
```

Reply:
This is great! I was not aware of this option